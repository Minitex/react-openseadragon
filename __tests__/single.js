import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import ReactOpenSeadragon from '../src/react-openseadragon-route';

jest.mock('react-dom', () => ({
    findDOMNode: () => ({
      scrollIntoView: jest.fn(),
    }),
  }),
);

jest.mock('react-lazy-load', () => 'LazyLoad');

describe('OpenSeadragon', () => {
  document.body.innerHTML = '<div id="page-search" />';

  it('should render correctly', () => {
    window.OpenSeadragon = jest.fn();
    window.OPENSEADRAGONVIEWER = {
      addHandler: () => false,
    };

    const methods = {
      addHandler: () => false,
      goToPage: () => false,
    };

    Object.defineProperty(window, 'OpenSeadragon', {
      value: jest.fn(() => methods),
    });

    const config = {
      basename: 'example/image',
      getImageURL: 'http://cdm16022.contentdm.oclc.org/utils/ajaxhelper',
      osdConfig: {
        defaultZoomLevel: 0,
        tileSources: [
          'https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json',
        ],
      },
      pages: [
        {
          id: 0,
          title: 'MLK',
          sidebarThumbnail: 'https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/full/100,/0/default.jpg',
          transcript: 'MLK',
          viewer: 'OSD_VIEWER',
          cdmCollection: 'mpls',
          cdmIdentifier: '3188',
          infoURL: 'https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json',
        },
      ],
    };

  const component = renderer.create(
    <MemoryRouter initialEntries={['/0']} initialIndex={1} >
      <ReactOpenSeadragon {...config} />
    </MemoryRouter>);
    const jsonComponent = component.toJSON();
    expect(jsonComponent).toMatchSnapshot();
  });
});
