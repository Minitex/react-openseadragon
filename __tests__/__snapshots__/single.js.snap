// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenSeadragon should render correctly 1`] = `
<div>
  <div>
    <div>
      <div
        className="osd"
      >
        <div
          className="row osd-row"
        >
          <div
            className="easer showViewer col-md-12"
          >
            <div>
              <div
                className="openseadragon"
                id="osd-viewer"
              >
                <ul
                  className="osd-toolbar"
                >
                  <li>
                    <a
                      id="zoom-in"
                    >
                      <i
                        className="glyphicon glyphicon-zoom-in"
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      id="zoom-out"
                    >
                      <i
                        className="glyphicon glyphicon-zoom-out"
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      id="reset"
                    >
                      <i
                        className="glyphicon glyphicon-home"
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      id="full-page"
                    >
                      <i
                        className="glyphicon glyphicon-resize-full"
                      />
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            className=" easer hideViewer col-md-12"
            id="text-viewer"
          />
          <div
            className="col-md-0 easer"
          >
            
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
